### PR TITLE
Fix name clashing of terminal and non-terminal rules

### PR DESCRIPTION
**Pull request by Pluralia**
Closes [#294](https://github.com/langium/langium/issues/294).

### CODE REVIEW

**Review by msujew — COMMENTED** [review]
Do you mind adding an automated test for this? I.e. create a grammar with a terminal/keyword and parser rule with the same name. Creating a parser for this grammar should not result in an error.

**Comment by msujew** [comment]
@Pluralia For the tests, you can reuse this code, which I wrote for another PR:

https://github.com/langium/langium/blob/81b0aff271609707f411aed2e466aea5a3189598/packages/langium/test/parser/langium-parser-builder.test.ts#L76-L95

File: packages/langium/test/parser/langium-parser-builder.test.ts [hunk 1]
```diff
@@ -73,6 +73,37 @@ describe('Predicated grammar rules', () => {
...
+        parserFromGrammar(parseResult.value);
+    });
+
+    test('Should have no lexer errors', () => {
+        expect(parseResult.lexerErrors).toEqual([]);
+    });
+
+    test('Should have no parser errors', () => {
+        expect(parseResult.parserErrors).toEqual([]);
+    });
```
Comment (msujew) [inline packages/langium/test/parser/langium-parser-builder.test.ts:103]: These tests don't actually do anything. Instead, the actual test should consist in the `parserFromGrammar` method not to throw:

```ts
test('Should not throw on parser validation', () => {
  expect(() => {
    parserFromGrammar(parseResult.value);
  }).not.toThrow();
});
```

Comment (Pluralia) [reply packages/langium/test/parser/langium-parser-builder.test.ts]: These tests fall, if I execute them on non-modified version and call `parserFromGrammar` [here](https://github.com/langium/langium/blob/0d5c796b2a9c4afa46121176b3344f1faf60a9a6/packages/langium/test/parser/langium-parser-builder.test.ts#L94), and I thought that they are working. Now I've got that any tests fall after `parserFromGrammar` executing, when it throws exeptions, so, my tests really do nothing. I rewrote them.

File: packages/langium/src/parser/langium-parser-builder.ts [hunk 2]
```diff
@@ -282,4 +284,4 @@ function wrap(ctx: RuleContext, method: Method, cardinality: Cardinality): Metho
     } else {
         throw new Error();
     }
-}
+}
```
Comment (msujew) [inline packages/langium/src/parser/langium-parser-builder.ts:287]: Nitpick: You removed the newline here

**Review by msujew — APPROVED** [review]
Looks good to me 👍

**Pull request merged**